util/lrucache/lrucache.go

```diff
@@ -110,12 +110,18 @@ func (c *Cache) Remove(key string) {
 	c.cache.Remove(key)
 }
 
-func (c *Cache) decreaseOnceFunc(rc *refCounter) func() {
+func (c *Cache) decreaseOnceFunc(rc *refCounter) func(bool) {
 	var once sync.Once
-	return func() {
+	return func(evict bool) {
 		c.mu.Lock()
 		defer c.mu.Unlock()
 		once.Do(func() { rc.dec() })
+		if evict {
```

Comment:
Seems that the decrement is locked behind a sync.Once but the evict isn't. So this means that `done(true)` can be called at anytime after `done(false)` and on that call it will do the eviction. If this is intentional I wonder if a separate cleanup function is the right call here. Having a `Close()` call that's functionally the same as `Done()` is a little strange, in particular because calling `Done()` signifies that we are done with the layer, so having a separate call that can also say the layer is done being used seems to be a strange pattern.